cpp/src_prims/linalg/batched/matrix.cuh

```diff
@@ -181,6 +180,35 @@ class Matrix {
     initialize(setZero);
   }
 
+  /**
+   * @brief Constructor that uses pre-allocated memory.
+   * 
+   * @param[in]  m            Number of rows
+   * @param[in]  n            Number of columns
+   * @param[in]  batch_size   Number of matrices in the batch
+   * @param[in]  cublasHandle cuBLAS handle
+   * @param[in]  allocator    Device memory allocator
+   * @param[in]  d_batches    Pre-allocated pointers array
+   * @param[in]  d_dense      Pre-allocated dense data array
```

Comment:
Yes, `d_batches` points at locations in the dense array separated by a fixed stride of `m * n`. This array of pointers is used by a few operations that don't have a strided version.

We could actually optimize a bit by doing a lazy computation: only compute these pointers when first needed, if ever. This is easy to add by modifying the wrapper that returns the pointer, and adding a boolean property to the class.

Reply:
Note that the kernel that fills this array of pointers accounts for 9% of the GPU work in the timeline for a non-seasonal case above.

Reply:
Thanks for the explanation. It is fine to pass the already initialzied list of pointers. But it would be good to describe the relationship between d_batches  and d_dense (m_batches/m_dense) somewhere in the docstring of the class.

Reply:
Ok, I will add that in the docstring.

The list of pointers is not yet initialized when passed to the constructor, it is initialized at construction. I take note to change this to a lazy approach in another PR.